spring-rabbit/src/main/java/org/springframework/amqp/rabbit/connection/SimpleConnection.java

```diff
@@ -71,15 +74,41 @@ public void close() {
 
 	@Override
 	public boolean isOpen() {
-		return this.delegate != null
-				&& (this.delegate.isOpen() || this.delegate.getClass().getSimpleName().contains("AutorecoveringConnection"));
+		if (this.delegate instanceof AutorecoveringConnection && !this.delegate.isOpen()) {
+			throw new AmqpException("Auto recovery connection is not currently open");
```

Comment:
Ok. Why just not return `this.delegate.isOpen()`?
At least the exception is thrown only if it isn't open.